front/lib/api/assistant/conversation.ts

```diff
@@ -1054,7 +1057,19 @@ export async function* postUserMessage(
   if (agentMessageRows.length !== agentMessages.length) {
     throw new Error("Unreachable: agentMessageRows and agentMessages mismatch");
   }
-
+  if (agentMessages.length > 0) {
+    for (const agentMessage of agentMessages) {
+      // Can't use Promise.all() here because populating the data
+      // on the first time would result in a very likely race condition
+      // that would result in counting multiple time each message.
+      // We'll need to add a lock here.
```

Comment:
Actually this can be solve by checking if the Redis keys exists using `incr` (vs `exists`).
`Incr` is thread safe, so if we get `1`, it means we need to populate, otherwise we can just insert the data. Order (obviously) does not matter.